product/en-US/development_guide/tools.xml

```diff
@@ -387,27 +383,29 @@ Normally you would leave participant log management to the transaction log that
                                 is not an exclusive resource,
                             </para>
                             <para>
-Normally you would leave participant log management to the transaction log that owns it or to the recovery system. However, this remove operation for participant logs is provided for those cases where you know it is safe to do so and, in the case of heuristically completed XA resources, you wish to trigger a forget call so that the XA resource vendors' logs are cleaned correctly. By default, if this forget call fails then the delete operation will fail. The system administrator may override this behaviour by setting a system property:
-                                <screen>ObjectStoreEnvironmentBean.ignoreMBeanHeuristics</screen> to the value true.
+Normally you would leave participant log management to the transaction log that owns it or to the recovery system. However, this remove operation for participant logs is provided for those cases where you know it is safe to do so and, in the case of heuristically completed XA resources, you wish to trigger a forget call so that the XA resource vendors' logs are cleaned correctly. By default, if this forget call fails then the delete operation will still succeed. The system administrator may override this behaviour by setting a system property:
+                                <screen>ObjectStoreEnvironmentBean.ignoreMBeanHeuristics</screen> to the value false.
                             </para>
                         </warning>
                     </listitem>
                 </varlistentry>
-                <varlistentry id="recover_transaction_jmx">
+                <varlistentry id="recover_transaction_participant_jmx">
                     <term>Recover a transaction.</term>
                     <listitem>
                         <para>
-                            Transaction participants supports recovery via the
+                            Transaction participants support recovery via the
                             <code>clearHeuristic</code>
                             operation.
                         </para>
                         <itemizedlist>
-                            <title>Recovery of Heuristic Transactions and Participants</title>
+                            <title>Recovery of Heuristic Participants</title>
                             <listitem>
                                 <para>
-                                    If the transaction's status is
-                                    <literal>HEURISTIC</literal>
-                                    , the clearHeuristic operation changes the state to
+                                    If the transaction participant's status is
+                                    <literal>HEURISTIC</literal> or
```

Comment:
When does a participant end up in this unknown heuristic state (ie it should be one of mixed, hazard, rollback or commit)?